api/src/main/java/org/openmrs/api/impl/EncounterServiceImpl.java

```diff
@@ -96,38 +126,20 @@ public Encounter saveEncounter(Encounter encounter) throws APIException {
 		
 		// if authenticated user is not supposed to edit encounter of certain type
 		if (!canEditEncounter(encounter, null)) {
```

Comment:
Why not move the whole block into failIfDeniedToEdit() instead of just moving APIException?